libs/SmartStore/SmartStore/Classes/NSData+SFAdditions.h

```diff
@@ -0,0 +1,37 @@
+//
+//  NSData+SFAdditions.h
+//  SmartStore
+//
+//  Created by Brianna Birman on 1/27/21.
+//  Copyright (c) 2021-present, salesforce.com, inc. All rights reserved.
+// 
+//  Redistribution and use of this software in source and binary forms, with or without modification,
+//  are permitted provided that the following conditions are met:
+//  * Redistributions of source code must retain the above copyright notice, this list of conditions
+//  and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright notice, this list of
+//  conditions and the following disclaimer in the documentation and/or other materials provided
+//  with the distribution.
+//  * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+//  endorse or promote products derived from this software without specific prior written
+//  permission of salesforce.com, inc.
+// 
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+//  IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+//  FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+//  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+//  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+//  WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+/** Provides MD5 conversion support.
+ */
+@interface NSData (SFSHA)
+
+/**Derives  a 16-byte  hex encoded string .
+ @return 16-byte  hex encoded string.
+ */
+-(NSString *)digest;
```

Comment:
Should this live in the SalesforceSDKCommon library instead?

Reply:
This should be completely removed in `9.1`. NCC keeps flagging it because of `MD5` usage. The only reason we're making it internal for now is to give us time to ensure that we have all the required upgrade steps in place.

Reply:
It was originally in core but I moved it here because SmartStore couldn't access it after I took the public API out, but if there's a better way to handle that I can switch

Reply:
I see. It makes sense to have it here because it needs to be internal.